### PR TITLE
Upgrade deployment to 2.2.x

### PR DESCRIPTION
**Pull request by Mohitv**
**Changelog**
- Add : Upgrade deployment of Hypeledger Fabric deployment to 2.2.x. The documentation is available in docs folder at /docs/source/operations/upgrade_fabricnetwork_1.4.x_2.2.2.md
 

**Reviewed by**
@sownak @jagpreetsinghsasan 

 

**Linked issue**
#1845

### CODE REVIEW

**Comment by jagpreetsinghsasan** [comment]
@Mohitv another huge commits.
Once you create a PR using a branch, please do not use that branch to do anymore work involving pushing of such commits. Received 81 mails for 81 commits

**Comment by Mohitv** [comment]
> @Mohitv another PR with huge number of commits. Once you create a PR using a branch, please do not use that branch to do anymore work involving pushing of such commits. Received 81 mails for 81 commits

@jagpreetsinghsasan I was not aware of this and was testing on my branch for the demo. If possible please cancel this PR and I will create a new

**Comment by sownak** [comment]
Please check https://github.com/hyperledger/bevel/blob/main/CONTRIBUTING.md. 
There is a conflict, that needs to be resolved.

File: platforms/hyperledger-fabric/configuration/roles/upgrade/orderer-binary/tasks/create_orderer_value.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    org_name: "{{ item.name | lower }}"
+    component_name: "{{ orderer.name }}-{{ org_name }}"
+    orderer_image: "hyperledger/fabric-orderer:{{ network.version }}"
+    alpine_image: "{{ docker_url }}/alpine-utils:1.0"
+    type: "orderers"
+    consensus: "{{component_services.consensus}}"
+    genesis: "{{ lookup('file', '{{ build_path }}/channel-artifacts/genesis.block.base64') }}"
+
+#Git Status : check if there is any files modified in the directory
+- name: Git Status
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/upgrade/orderer-binary/tasks/create_orderer_value.yaml:24]: This is not required. Also, I see you registered the output of this task to `git_result` and in the very next task also, the registered variable is `git_result` which is useless

File: platforms/hyperledger-fabric/configuration/roles/upgrade/peer-binary/tasks/create_peer_value.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    component_name: values-{{ peer.name }}
+    couchdb_image: "{{ couchdb_image_version[network.version] }}"
+    peer_image: "hyperledger/fabric-peer:{{ network.version }}"
+    alpine_image: "{{ docker_url }}/alpine-utils:1.0"
+    peer_name: "{{ peer.name }}"
+    peer_ns: "{{ namespace }}"
+    provider: "{{ network.env.proxy }}"
+
+#Git Status : check if there is any files modified in the directory
+- name: Git Status
```
Comment (jagpreetsinghsasan) [inline platforms/hyperledger-fabric/configuration/roles/upgrade/peer-binary/tasks/create_peer_value.yaml:25]: Same comment for the git status task as mentioned previously.

File: platforms/shared/configuration/roles/git_status/Readme.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+[//]: # (##############################################################################################)
```
Comment (jagpreetsinghsasan) [inline platforms/shared/configuration/roles/git_status/Readme.md:1]: Not required

File: platforms/shared/configuration/roles/git_status/tasks/main.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
+##############################################################################################
```
Comment (jagpreetsinghsasan) [inline platforms/shared/configuration/roles/git_status/tasks/main.yaml:1]: Not required

File: platforms/shared/configuration/roles/check/helm_component/tasks/main.yaml [hunk 5]
```diff
@@ -53,3 +53,4 @@
   delay: "{{ delay | default(30) }}"
   until: component_data.resources|length > 0
   when: component_type == "Pod"
+
```
Comment (jagpreetsinghsasan) [inline platforms/shared/configuration/roles/check/helm_component/tasks/main.yaml:56]: Newline at EOF

**Review by jagpreetsinghsasan — CHANGES_REQUESTED** [review]
There are repeating review comments (many places I havent even mention them as they are so many, but very small ones).
Also certain files like reset.sh, git_push task are empty which is required to be fixed.

File: platforms/hyperledger-fabric/configuration/roles/upgrade/add-app-channel-acls/tasks/main.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    state: absent
+
+# Create directory
+- name: Create build directory if it does not exist
+  file:
+    path: "{{ build_path }}/channel-artifacts/"
+    state: directory
+
+# Ensure channel-artifacts dir exists
+- name: Ensure channel-artifacts dir exists
```
[the remark this answers is not in the document]
Comment (Mohitv) [reply platforms/hyperledger-fabric/configuration/roles/upgrade/add-app-channel-acls/tasks/main.yaml]: I have kept only 2 tasks, First will remove and second will create a new folder. I have tested the first and it does not throw error if directory does not exists

File: platforms/hyperledger-fabric/configuration/roles/upgrade/add-app-channel-acls/tasks/main.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    src: "{{channel.acls}}"
+    dest: "{{ build_path }}/channel-artifacts/add_app_channel_acls.json"
+  when: acls_file.stat.exists
+
+# This task add default acls in app channels application group
+- name: Add add default acls in app channels application group
+  include_tasks: nested_main_add_app_channel_acls.yaml
+  vars:
+    channel_name: "{{ channel.channel_name | lower }}"
+  when: acls_file.stat.exists
```
[the remark this answers is not in the document]
Comment (Mohitv) [reply platforms/hyperledger-fabric/configuration/roles/upgrade/add-app-channel-acls/tasks/main.yaml]: done. here and all other instance mentioned in reviews

File: platforms/hyperledger-fabric/configuration/roles/upgrade/add-app-channel-acls/tasks/sign_block.yaml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  when: existing_cli.resources|length == 0
+
+# Start the cli using the value file created in the previous step
+- name: "Start cli"
+  shell: |
+    KUBECONFIG={{ org.k8s.config_file }} helm upgrade --install -f {{ build_path }}/{{ participant.name }}/existing_peer_cli_job.yaml {{ peer.name }}-{{ participant.name }}-cli {{playbook_dir}}/../../../{{org.gitops.chart_source}}/fabric_cli
+  when: existing_cli.resources|length == 0
+
+# waiting for fabric cli
+- name: "Check if fabric cli is present"
```
[the remark this answers is not in the document]
Comment (Mohitv) [reply platforms/hyperledger-fabric/configuration/roles/upgrade/add-app-channel-acls/tasks/sign_block.yaml]: done. here and all other instance mentioned in reviews below

**Comment by jagpreetsinghsasan** [comment]
Closing this PR as a new one has been created here #2069

**Pull request closed without merging**